src/options.ts

```diff
@@ -1,18 +1,50 @@
 export interface IVPCOptions {
     cidr: string;
     subnets: string[];
+    //Optional ivars to dictate if will use existing VPC 
+    //and subnets specified
+    vpcId: string;
+    securityGroupIds: string[];
+    subnetIds: string[];
 }
 
 export interface IServiceProtocolOptions {
     protocol: "HTTP" | "HTTPS";
     certificateArns?: string[]; // needed for https
 }
 
+enum AutoScalingMetricType {
+    ALBRequestCountPerTarget,
+    AppStreamAverageCapacityUtilization,
+    ComprehendInferenceUtilization,
+    DynamoDBReadCapacityUtilization,
+    DynamoDBWriteCapacityUtilization,
+    EC2SpotFleetRequestAverageCPUUtilization,
+    EC2SpotFleetRequestAverageNetworkIn,
+    EC2SpotFleetRequestAverageNetworkOut,
+    ECSServiceAverageCPUUtilization,
+    ECSServiceAverageMemoryUtilization,
+    LambdaProvisionedConcurrencyUtilization,
+    RDSReaderAverageCPUUtilization,
+    RDSReaderAverageDatabaseConnections,
+    SageMakerVariantInvocationsPerInstance
+};
+
+export interface IServiceAutoScalingOptions {
+    min?: number; //default to 1
+    max?: number; //default to 1
+    metric: AutoScalingMetricType;
+    cooldown?: number; //defaults to 30
+    cooldownIn?: number; //defaults to cooldown but has priority over it
+    cooldownOut?: number; //defaults to cooldown but has priority over it
+    targetValue: number;
+}
+
 export interface IServiceOptions {
     name: string;
     cpu: number;
     memory: number;
-    port: number;
+    port?: number;
```

Comment:
Probably should add a comment saying it defaults to a randomly generated port.

Reply:
Done :)